lib/util.js

```diff
@@ -31,8 +31,35 @@ function isEmptyObject (obj) {
   return true
 }
 
+function getProperty(stylelint, prop) {
+  var stylelintProperty = null
+  if (stylelint && stylelint[prop]) {
+    stylelintProperty = stylelint[prop]
+    if(typeof stylelintProperty == 'object') {
+      stylelintProperty = stylelintProperty[0]
+    }
+  }
+
+  return stylelintProperty
+}
+
+function getOptions(stylelint, prop, option) {
+  var stylelintProperty = []
+  if (stylelint && stylelint[prop]) {
+    stylelintProperty = stylelint[prop]
+    if(typeof stylelintProperty == 'object' && stylelintProperty.length > 1) {
+      stylelintProperty = stylelintProperty[1][option]
```

Comment:
The rule 'at-rule-empty-line-before' can take optional parameters that get passed in via the stylelintrc. E.g. "at-rule-empty-line-before": ["always", { except: ["blockless-group"] }]

So stylelintProperty[1] is grabbing the object: { except: ["blockless-group"] } and [option] then takes the value of except which in this case is 'blockless-group'.

https://github.com/stylelint/stylelint/blob/master/src/rules/at-rule-empty-line-before/README.md